OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java

```diff
@@ -2001,18 +2084,53 @@ public void run() {
       });
    }
 
-   // Called when receiving FCM/ADM message after it has been displayed.
-   // Or right when it is received if it is a silent one
-   //   If a NotificationExtenderService is present in the developers app this will not fire for silent notifications.
-   static void handleNotificationReceived(JSONArray data, boolean displayed) {
-      OSNotificationOpenResult openResult = generateOsNotificationOpenResult(data, displayed);
-      if(trackFirebaseAnalytics != null && getFirebaseAnalyticsEnabled())
-         trackFirebaseAnalytics.trackReceivedEvent(openResult);
+   /**
+    * Called when receiving FCM/ADM message after it has been displayed.
+    * Or right when it is received if it is a silent one
+    *   If a NotificationExtenderService is present in the developers app this will not fire for silent notifications.
+    */
+   static void handleNotificationReceived(OSNotificationGenerationJob notifJob, boolean displayed) {
+      int androidNotificationId = notifJob.getAndroidId();
+
+      try {
+         JSONObject jsonObject = new JSONObject(notifJob.jsonPayload.toString());
+         jsonObject.put("notificationId", androidNotificationId);
```

Comment:
does it make sense to start having this string as constant?

Reply:
We shouldnt have any string constants in `OneSignal.java`
I will eventually clean this up, but for now we will leave it like this

Reply:
I take back the previous comment, Josh made a constant for this and we are suing it now